src/index.ts

```diff
@@ -1,93 +1,155 @@
-import * as core from '@actions/core';
-import * as github from '@actions/github'
-import { LinearClient } from '@linear/sdk'
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { LinearClient } from "@linear/sdk";
 import { isNil } from "lodash";
 
 const PR_TITLE_UPDATE_KEYWORD = "x";
 
 type PullRequest = {
-    number: number;
-    title: string;
-    head: {
-        ref: string;
-        repo: {
-            name: string;
-            owner: {
-                login: string;
-            }
-        } | null,
-    },
-}
+  number: number;
+  title: string;
+  body: string | null;
+  head: {
+    ref: string;
+    repo: {
+      name: string;
+      owner: {
+        login: string;
+      };
+    } | null;
+  };
+};
 
-export const getIssueIdFromBranchName = (branchName: string): string | null => {
-    const match = branchName.match(/^feature\/([a-z]+\-\d+)\-/);
-    if (isNil(match)) return null;
-    const branchStoryId = match[1];
-    return branchStoryId.toUpperCase();
-}
+export const getLinearIssueIds = (pullRequest: PullRequest) => {
+  const issueIds: string[] = [];
+
+  // From the branch name
+  const branchName = pullRequest.head.ref;
+  let match = branchName.match(/^([a-z]+\-\d+)\-/);
+  if (!isNil(match)) issueIds.push(match[1].toUpperCase());
+
+  // From the PR body
+  const body = isNil(pullRequest.body) ? "" : pullRequest.body;
+  const bodyRegex = /Fixes ([a-z]+\-\d+)|Resolves ([a-z]+\-\d+)/gi;
+  match = bodyRegex.exec(body);
+  while (match != null) {
+    issueIds.push(match[1] || match[2]);
+    match = bodyRegex.exec(body);
+  }
+
+  return issueIds;
+};
+
+const getTitleFromIssueId = async (
+  linearClient: LinearClient,
+  pullRequest: PullRequest,
+  issueId: string
+) => {
+  if (pullRequest.title != PR_TITLE_UPDATE_KEYWORD) {
+    core.info(`PR title isn't set to keyword '${PR_TITLE_UPDATE_KEYWORD}'.`);
+    return pullRequest.title;
+  }
+
+  const issue = await linearClient.issue(issueId);
+  const parentIssue = await issue.parent;
```

Comment:
Do we need the await here? 😕

Reply:
Seems like it. Otherwise I get this error:

![image](https://user-images.githubusercontent.com/1542063/171283838-28369a64-3715-4d47-8cb6-f7d9568f8e2f.png)